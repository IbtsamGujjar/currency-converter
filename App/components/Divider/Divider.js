import React from 'react';
import PropTypes from 'prop-types';

import { View, StyleSheet } from 'react-native';

import { Colors } from '../../../constants/colors';

const styles = StyleSheet.create({
  divider: {
    backgroundColor: Colors.border,
    height: StyleSheet.hairlineWidth,
  },
});

export const Divider = ({ style }) => {
  return <View style={{ ...styles.divider, ...style }} />;
};

Divider.defaultProps = {
  style: {},
};

Divider.propTypes = {
  style: PropTypes.objectOf(PropTypes.any),
};
