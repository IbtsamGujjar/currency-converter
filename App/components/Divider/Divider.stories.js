import React from 'react';
import { storiesOf } from '@storybook/react-native';

import { Divider } from './Divider';
import { CenterView } from '../../../storybook/stories/decorator';

storiesOf('Divider', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('Basic', () => <Divider style={{ backgroundColor: 'red' }} />);
