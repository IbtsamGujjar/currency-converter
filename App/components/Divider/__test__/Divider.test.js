import React from 'react';
import renderer from 'react-test-renderer';
import { Divider } from '../Divider';

test('Divider Snapshot', () => {
  const snapshot = renderer.create(<Divider />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
