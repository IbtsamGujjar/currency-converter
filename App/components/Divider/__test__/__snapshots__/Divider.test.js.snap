// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e2e2e2",
      "height": 0.5,
    }
  }
/>
`;
